compiler/rustc_errors/src/emitter.rs

```diff
@@ -2054,8 +2054,17 @@ fn num_decimal_digits(num: usize) -> usize {
     MAX_DIGITS
 }
 
+const REPLACEMENTS: &[(char, &str)] = &[
+    ('\t', "    "),
+    ('\u{200D}', ""), // Replace ZWJ with nothing for consistent terminal output of grapheme clusters.
+];
+
 fn replace_tabs(str: &str) -> String {
```

Comment:
This function should probably be renamed, to something like `normalize_whitespace`, not that its no longer just tabs.